### PR TITLE
new tool: jvarkit/vcfpolyx

### PR DESCRIPTION
**Pull request by lindenb**
this PR is a new module for `jvarkit/vcfpolyx` (I want to fix https://github.com/nf-core/sarek/issues/904 )

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [X] `nf-core modules test <MODULE> --profile singularity`
    - [X] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/jvarkit/vcfpolyx/environment.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/environment-schema.json
+name: "jvarkit_vcfpolyx"
+channels:
+  - bioconda
```
Comment (maxulysse) [inline modules/nf-core/jvarkit/vcfpolyx/environment.yml:5]: just bioconda?

Comment (lindenb) [reply modules/nf-core/jvarkit/vcfpolyx/environment.yml:5]: @maxulysse isn't there a problem with conda that is now commercial ?  and I there is currently a problem with environment.yml...
```
: 'conda-forge' is │ │                   │                            │ not one of ['bioconda']. 
```

**Comment by lindenb** [comment]
@nf-core-bot fix linting pretty please 🙏

**Comment by lindenb** [comment]
@nf-core-bot fix linting pretty please 🙏

**Comment by lindenb** [comment]
@maxulysse back to you ! :-P

File: modules/nf-core/jvarkit/vcfpolyx/main.nf [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    extension =     args3.contains("--output-type b") || args3.contains("-Ob") ? "bcf.gz" :
+                    args3.contains("--output-type u") || args3.contains("-Ou") ? "bcf" :
+                    args3.contains("--output-type z") || args3.contains("-Oz") ? "vcf.gz" :
+                    args3.contains("--output-type v") || args3.contains("-Ov") ? "vcf" :
+                    "vcf"
+
+    if ("$vcf" == "${prefix}.${extension}") error "Input and output names are the same, set prefix in module configuration to disambiguate!"
+    """
+    set -o pipefail
```
Comment (maxulysse) [inline modules/nf-core/jvarkit/vcfpolyx/main.nf:44]:
```suggestion
```

I don't think we need that here as it should be handled by Nextflow already

File: modules/nf-core/jvarkit/vcfpolyx/meta.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      type: map
+      description: |
+        Groovy Map containing fasta information
+
+  - meta2:
+      type: map
+      description: |
+        Groovy Map containing fasta.fai information
+
+  - meta3:
```
Comment (nvnieuwk) [inline modules/nf-core/jvarkit/vcfpolyx/meta.yml:43]: Can you add meta4 too?

File: modules/nf-core/jvarkit/vcfpolyx/main.nf (hunk 2 above)
Comment (lindenb) [reply modules/nf-core/jvarkit/vcfpolyx/main.nf:44]: > it should be handled by Nextflow already

are you sure about this ? e.g: https://github.com/nextflow-io/nextflow/discussions/3792

Comment (maxulysse) [reply modules/nf-core/jvarkit/vcfpolyx/main.nf:44]: this is in the TEMPLATE: https://github.com/nf-core/tools/blob/master/nf_core/pipeline-template/nextflow.config#L224

Comment (maxulysse) [reply modules/nf-core/jvarkit/vcfpolyx/main.nf:44]: so yeah, Nextflow will handle it for you if it's specified in your config, which is for nf-core pipelines

Comment (lindenb) [reply modules/nf-core/jvarkit/vcfpolyx/main.nf:44]: @maxulysse oh ! ok, good to know !

**Comment by lindenb** [comment]
@maxulysse done :-)

File: modules/nf-core/jvarkit/vcfpolyx/main.nf [hunk 4]
```diff
@@ -0,0 +1,70 @@
+/**
+ * JVARKIT_VCFPOLYX
+ * Author: Pierre Lindenbaum PhD
+ * vcfpolyx is a sub-command of the jvarkit package. It is used to annotate a vcf for the poly-x repeats
+ */
```
Comment (maxulysse) [inline modules/nf-core/jvarkit/vcfpolyx/main.nf:5]:
```suggestion
```
I'd say this info is in the meta.yml, no?

**Review by maxulysse — APPROVED** [review]

**Pull request merged**